### PR TITLE
Master+som1 sdmmc0

### PR DESCRIPTION
**Pull request by claudiubeznea**
Hi,

It has been reported in [1] that starting w/ v21.02.0 SAMA5D27 SOM1 EK fails to boot.
It is due to the fact that default booting media has been changed in b/w w/o any reason.
Revert back to SDMMC0 as this is the default booting media for this board in all the
other Microchip supported distribution for this board (Buildroot, Yocto Project).

@DennisDNguy: could you check if this fixes the problem on your side as I don't have
a board available to check it.

Thank you,
Claudiu Beznea

[1] https://github.com/openwrt/openwrt/issues/10276

### CODE REVIEW

File: package/boot/uboot-at91/Makefile [hunk 1]
```diff
@@ -107,6 +107,12 @@ define U-Boot/sama5d4_xplained_nandflash
   BUILD_DEVICES:=microchip_sama5d3-xplained
 endef
 
+define U-Boot/sama5d27_som1_ek_mmc
+  NAME:=SAMA5D27 SOM1 Ek (SDCard0)
+  BUILD_SUBTARGET:=sama5
+  BUILD_DEVICES:=microchip_sama5d27-som1-ek
+endef
+
 define U-Boot/sama5d27_som1_ek_mmc1
```
Comment (hauke) [inline package/boot/uboot-at91/Makefile:116]: `sama5d27_som1_ek_mmc1` is not referenced any more, you can delete it here.

File: package/boot/at91bootstrap/Makefile [hunk 2]
```diff
@@ -193,7 +193,7 @@ AT91BOOTSTRAP_TARGETS := \
 	sama5d4_xplainednf_uboot_secure \
 	sama5d4_xplaineddf_uboot_secure \
 	sama5d4_xplainedsd_uboot_secure \
-	sama5d27_som1_eksd1_uboot \
```
Comment (hauke) [inline package/boot/at91bootstrap/Makefile:196]: You can probably remove `sama5d27_som1_eksd1_uboot` too.

File: target/linux/at91/image/sama5.mk [hunk 3]
```diff
@@ -19,13 +19,13 @@ define Build/at91-sdcard
...
-      $(if $(findstring sama5d4-xplained,$@), \
-          mcopy -i $@.boot \
+  $(if $(findstring sama5d4-xplained,$@), \
+	  mcopy -i $@.boot \
               $(BIN_DIR)/at91bootstrap-$(DEVICE_DTS:at91-%=%)sd_uboot_secure/at91bootstrap.bin \
               ::BOOT.bin,
           mcopy -i $@.boot \
               $(BIN_DIR)/at91bootstrap-$(DEVICE_DTS:at91-%=%)sd_uboot/at91bootstrap.bin \
-              ::BOOT.bin))
+              ::BOOT.bin)
```
Comment (hauke) [inline target/linux/at91/image/sama5.mk:28]: This looks like a fix for a bug introduced in the previous commit, please move it to the previous commit.

**Comment by DennisDNguy** [comment]
On July 31, 2022, I checked out the master branch git clone https://git.openwrt.org/openwrt/openwrt.git and built the AT19 Microchip SAMA5D27 SOM1 Ek :
I got the following error messages:
RomBOOT
AT91Bootstrap 4.0.3 (2022-07-28 15:46:33)
SDHC: fix in place for SAMA5D2 SiP VDDSDMMC over-consumption errata
SD/MMC: Image: Read file u-boot.bin to 0x23f00000
MMC: ADMA supported
SDHC: Error: No Card Inserted
*** FATFS: f_open, filename: [u-boot.bin]: error
SD/MMC: Failed to load image

**Comment by claudiubeznea** [comment]
@DennisDNguy : did you managed to apply the patches on this pull request on top of master?

**Comment by claudiubeznea** [comment]
@hauke: sure, I'll adapt the pull request with your comments, thank you!

**Comment by hauke** [comment]
This was superseded by #10575.

**Pull request closed without merging**